src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/ParsingSimpleTypeModelBinder.cs

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Globalization;
+using System.Runtime.ExceptionServices;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Mvc.ModelBinding.Binders
+{
+    /// <summary>
+    /// An <see cref="IModelBinder"/> for decimal, double, float, and their <see cref="Nullable{T}"/> wrappers.
+    /// </summary>
+    public class ParsingSimpleTypeModelBinder : IModelBinder
```

Comment:
It might be worth separating this into an implementation-per-type so an enterprising user could configure them separately (like passing in different number styles values). 

Then the problem of naming is much easier.